CMakeLists.txt

```diff
@@ -133,8 +133,8 @@ add_subdirectory(${xgboost_SOURCE_DIR}/plugin)
 add_subdirectory(${xgboost_SOURCE_DIR}/src)
 set(XGBOOST_OBJ_SOURCES "${XGBOOST_OBJ_SOURCES};$<TARGET_OBJECTS:objxgboost>")
 
-#-- Shared library
-add_library(xgboost SHARED ${XGBOOST_OBJ_SOURCES})
+#-- library
+add_library(xgboost ${XGBOOST_OBJ_SOURCES})
```

Comment:
CI is breaking because with this change static lib is generated by default. Is it possible to make SHARED lib a default?

Reply:
Ah that's interesting on my local, shared was generated by default. I'll investigate some more and follow up.

Reply:
We could declare `BUILD_SHARED_LIBS` as an option and give false as the default value.

Reply:
Yup, just made that change. Hopefully it passes now.